lib/sender.js

```diff
@@ -81,14 +86,24 @@ class Sender {
 
     if (!options.mask) return [target, data];
 
-    randomFillSync(mask, 0, 4);
+    let mask = _mask;
+
+    if (options.generateMask) {
+      mask = options.generateMask();
```

Comment:
This generates a new Buffer object on every send, instead it should modify existing buffer

Reply:
Yeah it's a totally invalid argument to make that checking if an uint32_t is 0 (one jz instruction in constant time) would be too costly, while having an API that does garbage collected memory allocations in the hot path (literally gazillions of instructions in nondeterministic time).

Reply:
That's like arguing about the weight of a toothpick, when moving a house.

Reply:
Not necessarily.

Reply:
It makes the config setting awkward to require maintaining the buffer outside the setting, it would be more convenient and succinct to pass the buffer to the function: 
```javascript
new Websocket(url, { generateMask: buf => require('crypto').randomFillSync(buf) } )
```
and then allow passing a Buffer instead of a function if you don't plan on updating it, if you want to optimize:
```javascript
new Websocket(url, { generateMask: Buffer.alloc(4) } )
```

Reply:
> it would be more convenient and succinct to pass the buffer to the function.

The only reason why I did not do that is that the buffer is shared between `WebSocket` instances and this might generate confusion even it is documented because people do not read.

Reply:
Does it need to be shared?

Reply:
No, it only saves some bytes.

Reply:
If you want to use the clear mask you could pass an empty function, which will keep the Buffer at its default 0 value:
```javascript
new Websocket(url, { generateMask: buf => { } } )
```

Reply:
> If you want to use the clear mask you could pass an empty function, which will keep the Buffer at its default 0 value

This is actually what made me use a function that returns a `Buffer`, because I wrote the test include here that way, and it was failing due to https://github.com/websockets/ws/pull/1990#discussion_r772217927.

Reply:
I think it is ok to have a mask Buffer per WebSocket connection, vs sharing 1 Buffer between all WebSocket connections

Reply:
It requires more changes because I don't want to add a mask buffer also for server clients and currently the `Sender` is server/client agnostic.

Reply:
There is also the static `Sender.frame()` method. It gets kinda messy.

Reply:
You only need to create an extra Buffer if the Sender has the config option `generateMask` set, otherwise you use the shared buffer

Reply:
See https://github.com/websockets/ws/pull/1990/commits/8ab0db4fa664ac8049db2ccef80e94da73b67944

Reply:
Looks like exactly what is needed for that, this would be a great feature nice work